### ISSUE TITLE
Rethink the Scratch Addons mission statement

### ISSUE THREAD

**Issue by lisa-wolfgang** [ScratchAddons/ScratchAddons#5211]
Our current mission statement dates back to the early days of the extension, when we still mostly adapted old userstyles and userscripts:

> Scratch Addons is a browser extension that provides new features and themes for the Scratch website and project editor. It has a mission to collect, archive, and put into use userstyles and userscripts, developed by several members of the Scratch community.

I proposed an updated mission statement as part of the website FAQ revision done a couple months ago:

> Scratch Addons combines new and existing Scratch extensions, userscripts, and userstyles for the Scratch website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for the development of Scratch modifications.

We should discuss how to update our mission statement to best reflect our goals for the future.

**Comment by Secret-chest** [comment]
Do something about WL, Apple and mxmou rejecting customisation. (Get them a job at Apple)

**Comment by mxmou** [comment]
> for the development of Scratch modifications.

Scratch modifications are modified versions of the Scratch source code. Maybe "new features for Scratch"?

**Comment by WorldLanguages** [comment]
> The mission is to provide a centralized, up-to-date platform for the development of Scratch modifications.

"Scratch modifications": can be confused with Scratch editor forks, such as TurboWarp (as mxmou said)
"Scratch extensions": can be confused with extensions such as Pen, Lego WeDo

Maybe we can use a phrase such as "community-developed (new) features".

**Comment by lisa-wolfgang** [comment]
So perhaps something like this would be more appropriate:

> Scratch Addons combines new and existing Scratch extensions, userscripts, and userstyles for the Scratch website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for community development of features and themes for Scratch.

**Comment by Hans5958** [comment]
I guess there should be a shorter version for some cases (like the tagline on the index, below the "All-in-one extension...")

**Comment by lisa-wolfgang** [comment]
I think we can keep the existing tagline.

> "All-in-one browser extension for Scratch"

It still makes plenty of sense as a descriptor for the current state of the extension.

**Comment by lisa-wolfgang** [comment]
Sorry, I just realized that I misread your comment. Depending on the context, I think we could omit either the first or last sentence, since the first is more user-focused and the last is more developer-focused.

**Comment by WorldLanguages** [comment]
> I guess there should be a shorter version for some cases (like the tagline on the index, below the "All-in-one extension...")

I agree on using only the first sentence, specifically in the tagline on the index

**Comment by penguinmoose** [comment]
@lisa-wolfgang I think "new and existing scratch extensions" may be misleading because Scratch does not have any browser extensions that provide features we have, and we are not creating new Scratch editor extensions (although we could). I think this first sentence would be more appropriate:

Scratch Addons combines _additional scratch features_, userscripts, and userstyles for the Scratch website and project editor into one easy-to-access and configurable browser extension.

**Comment by WorldLanguages** [comment]
Pinning to see if there's more discussion to be had, or if we're done.

**Comment by WorldLanguages** [comment]
Note that the following sentence, which is currently on the README, would also be replaced by this new mission statement:

> Scratch Addons is a WebExtension (supports both Chrome and Firefox). Scratch Addons' mission is to combine all existing Scratch extensions, userscripts and userstyles into a single easy-to-access place, while still letting users choose which ones to enable.

**Comment by Secret-chest** [comment]
> @lisa-wolfgang I think "new and existing scratch extensions" may be misleading because Scratch does not have any browser extensions that provide features we have, and we are not creating new Scratch editor extensions, like WorldLanguages said (although we could). I think this first sentence would be more appropriate:
> 
> Scratch Addons combines _additional community-developed scratch features_, userscripts, and userstyles for the Scratch website and project editor into one easy-to-access and configurable browser extension.
> 
> Also, I think only the first sentence is necessary, although this makes it more of a summary of what Scratch Addons is, rather than our mission/goals.

Features are either userscripts or userstyles. Maybe just say features and themes.

**Comment by WorldLanguages** [comment]
Many alternatives were proposed here, so it can get confusing.
Reminder that the currently suggested version is available at the top of the original issue. Any proposed modifications should be based off of it.

**Comment by WorldLanguages** [comment]
> I think "new and existing scratch extensions" may be misleading because Scratch does not have any browser extensions that provide features we have, and we are not creating new Scratch editor extensions, like WorldLanguages said (although we could).

I don't think this is a big deal.
The "all-in-one browser extension for Scratch" description will stay, so I think it will be clear enough.
Saying "Scratch extension" is less confusing than "Scratch modifications", which was the wording in the original suggestion.

**Comment by penguinmoose** [comment]
> > @lisa-wolfgang I think "new and existing scratch extensions" may be misleading because Scratch does not have any browser extensions that provide features we have, and we are not creating new Scratch editor extensions, like WorldLanguages said (although we could). I think this first sentence would be more appropriate:
> > Scratch Addons combines _additional community-developed scratch features_, userscripts, and userstyles for the Scratch website and project editor into one easy-to-access and configurable browser extension.
> > Also, I think only the first sentence is necessary, although this makes it more of a summary of what Scratch Addons is, rather than our mission/goals.
> 
> Features are either userscripts or userstyles. Maybe just say features and themes.

Agree, that would make the statement understandable by more people.

**Comment by WorldLanguages** [comment]
> Agree, that would make the statement understandable by more people.

Hmm, I'm not convinced. "Scratch extensions, userscripts, and userstyles" can still be understood even if you don't know what userscripts and userstyles are. "Features and themes" doesn't mean much, anything could be considered a "feature"...

**Comment by DNin01** [comment]
> > Agree, that would make the statement understandable by more people.
> 
> Hmm, I'm not convinced. "Scratch extensions, userscripts, and userstyles" can still be understood even if you don't know what userscripts and userstyles are. "Features and themes" doesn't mean much, anything could be considered a "feature"...

Enhancement?

**Comment by WorldLanguages** [comment]
IMO "extensions, userscripts and userstyles" is fine.

**Comment by Hans5958** [comment]
I think the statement of "combines new and existing..." is the one is confusing. Is it refering to the old userscript/userstyles combined with the new, fresh ideas, or is it about the offered userscript/userstyles augments what features Scratch offers, or is there any other intepretation?

If it is the latter, then the word "additional" can be used, like "adds additional features." Notice the word "feature" can be used here generally to refer the existing features inside the extension.

**Comment by WorldLanguages** [comment]
@Hans5958 In case you're suggesting a change, please re-write the whole sentence you're proposing to change.

> Is it refering to the old userscript/userstyles combined with the new, fresh ideas

That's how I understood it. Right now, it doesn't mean much because most addons are created with SA in mind, but historically one of the goals of SA was to offer all features that already existed and could be installed through other means.

**Comment by penguinmoose** [comment]
> > Agree, that would make the statement understandable by more people.
> 
> Hmm, I'm not convinced. "Scratch extensions, userscripts, and userstyles" can still be understood even if you don't know what userscripts and userstyles are. "Features and themes" doesn't mean much, anything could be considered a "feature"...

That's a good point. We can keep userscripts and userstyles.

**Comment by penguinmoose** [comment]
@lisa-wolfgang I'm a bit confused about what "_new and existing_ Scratch extensions, userscripts, and userstyles" means. We are not combining and centralizing userscripts and userstyles that already exist - we are creating our own.

**Comment by mxmou** [comment]
> @lisa-wolfgang I'm a bit confused about what "_new and existing_ Scratch extensions, userscripts, and userstyles" means. We are not combining and centralizing userscripts and userstyles that already exist - we are creating our own.

Some SA features (discuss button, developer tools, Scratch 2.0 → 3.0, customizable quotes and code blocks) were originally separate extensions or userscripts.

**Comment by WorldLanguages** [comment]
The word "existing" has a historical meaning that indicates one of the original goals of SA. Personally I think we should keep it, but it's true that it has little meaning today and could be potentially confusing. Still, I don't think it's worth to remove the word just to clear up any potential small confusion.

**Comment by BroJac5246** [comment]
> The word "existing" has a historical meaning that indicates one of the original goals of SA. Personally I think we should keep it, but it's true that it has little meaning today and could be potentially confusing. Still, I don't think it's worth to remove the word just to clear up any potential small confusion.

I agree, plus it's not like porting existing ones is banned.

**Comment by Secret-chest** [comment]
I think the first sentence from the old one is good.

**Comment by Hans5958** [comment]
If it is the former, then I think it is fine. I would say to use the term "old and new," but I'm still indecisive here.

> Scratch Addons combines old and new [...] for...

Probably a better improvement would be to simplify the phrase "Scratch extensions, userscripts, and userstyles" to something like "enhancements" (also suggested by @DNin01). In fact, that's the E in RES (Reddit Enhancement Suite). I don't think the word "Scratch" is needed since there is "...the Scratch website..." right after it.

> Scratch Addons combines new and existing enhancements for the Scratch website and project editor into...

**Comment by WorldLanguages** [comment]
> I would say to use the term "old and new"

I don't think that term sounds good.

**Comment by WorldLanguages** [comment]
> > Scratch Addons combines new and existing enhancements for the Scratch website and project editor into...

Hmm, I've reconsidered and I think that could be better. No need to use the terms "userscript" and "userstyle" when talking to end-users.

It would look like this:
![image](https://user-images.githubusercontent.com/17484114/205742612-af53c004-8a0b-433b-a715-60620d875e2d.png)

**Comment by lisa-wolfgang** [comment]
I think simplifying to "enhancements" is a good idea, but I'd be sure to include the word "themes" as well, since that's not something that obviously falls under "enhancements."

**Comment by WorldLanguages** [comment]
@lisa-wolfgang Could you write a proposal (write the full sentence again) so we can vote on it through GitHub reactions? I think we could discuss this forever, that's not the idea

**Comment by Hans5958** [comment]
> I'd be sure to include the word "themes" as well, since that's not something that obviously falls under "enhancements."

How come this isn't some kind of enhancement? An additional option for styles is an enhancement as it is an additional option.

In fact, this is a mission statement. That additional detail can be out outside the mission statement, such as the section after it.

**Comment by lisa-wolfgang** [comment]
I'm realizing that the word "enhancements" is actually problematic -- it implies an improvement, which is subjective. We are obviously biased towards thinking positively of our "enhancements," of course, but I don't think it's a good idea to include opinionated words like this in a mission statement. The original wording, "modification," is an example of a word with unbiased meaning. We say "features and themes" in the second sentence, so that might be a better alternative for the first sentence.

**Comment by lisa-wolfgang** [comment]
@WorldLanguages I updated the original message.

**Comment by Samq64** [comment]
Plus not everything is an enhancement, for example hide tutorial button or semicolon glitch.

**Comment by WorldLanguages** [comment]
New proposed mission statement: (edited Dec 6th - "Scratch extensions, userscripts and userstyles" has been changed for "features and themes")

> Scratch Addons combines new and existing features and themes for the Scratch website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for community development of new features and themes for Scratch.

Please upvote/downvote

**Comment by WorldLanguages** [comment]
Oh wow, 15 upvotes silently went by after 3 days. No downvotes, except mine (I added both reactions to make it easier for people to vote).
I'll send a pull request on Lisa's behalf, because I think she's a bit busy this week. The "description" field of `manifest.json` will be changed for v1.30.0, and the ScratchAddons.com website should update at the same time that version is released (or possibly ASAP - doesn't matter much)

**Comment by WorldLanguages** [comment]
We overlooked the fact Chrome and Firefox recommend the description shouldn't be longer than 132 characters:
![image](https://user-images.githubusercontent.com/17484114/206877693-24b98e07-d87a-4873-901c-4d89a70281cf.png)

I'm not sure if it's just a recommendation, or if store review processes will reject it for being too long and we'll be forced to change it. I'll do some research now.

163 characters: `Scratch Addons combines new and existing features and themes for the Scratch website and project editor into one easy-to-access and configurable browser extension.`

**Comment by WorldLanguages** [comment]
Current description (v1.29.0) is 108 characters long (`Scratch Addons provides customizable features and themes for the scratch.mit.edu website and project editor.`).

A simple way to get the current description shorter (only for the `manifest.json` description) would be removing the last part ("into one easy-to-access and configurable browser extension").

Proposal (104 chars): `Scratch Addons combines new and existing features and themes for the Scratch website and project editor.`
Unfortunately, the word "configurable" and "customizable" are nowhere to be seen here.

**Comment by WorldLanguages** [comment]
This would only affect:

1. The extension description when visiting chrome://extensions and its equivalent in other browsers
2. The first paragraph of the extension description on the Chrome Web Store. For consistency, it would also affect the other 2 extension stores, but we are not forced to, unlike the CWS.

**Comment by DNin01** [comment]
How about this for the first paragraph?
90 chars - `Provides many customizable features and themes for the Scratch website and project editor.`

We could talk about the mission later in the extension store descriptions. For example:
> Provides many customizable features and themes for the Scratch website and project editor.
>
> Our mission is to provide a centralized, up-to-date platform for community development of new features and themes for Scratch.

**Comment by WorldLanguages** [comment]
> We could talk about the mission later in the extension store descriptions

I don't think we planned to include the mission in the extension store descriptions. It fits better on the website.

**Comment by CST1229** [comment]
> **Proposal (104 chars):** `Scratch Addons combines new and existing features and themes for the Scratch website and project editor.`

I'd say the customizability should be highlighted instead of the extension having both new and existing features.

**Comment by Weredime** [comment]
> `The feature-packed browser extension for the Scratch website and project editor.`

...might be reasonable for the tagline.
It should be short and simple. It shouldn't have too many characters.

**Comment by WorldLanguages** [comment]
We already have a tagline: `All-in-one browser extension for Scratch`. We could decide to change it, or keep it and use it as the extension description.

> I'd say the customizability should be highlighted instead of the extension having both new and existing features

And how would you do that?

**Comment by mxmou** [comment]
> > I'd say the customizability should be highlighted instead of the extension having both new and existing features
> 
> And how would you do that?

Probably by keeping the current description.

**Comment by DNin01** [comment]
> We already have a tagline: `All-in-one browser extension for Scratch`. We could decide to change it, or keep it and use it as the extension description.

Maybe we could make that the first paragraph, and then use the next paragraphs to further describe the extension?

**Comment by Secret-chest** [comment]
> And how would you do that?

Replace "new and existing" with "customisable"

**Comment by WorldLanguages** [comment]
> Replace "new and existing" with "customisable"

So that would be exactly 100 characters for the "description" field of `manifest.json`:
`Scratch Addons combines customizable features and themes for the Scratch website and project editor.`

**Comment by DNin01** [comment]
> > Replace "new and existing" with "customisable"
> 
> So that would be exactly 100 characters for the "description" field of `manifest.json`:
> `Scratch Addons combines customizable features and themes for the Scratch website and project editor.`

I think this might not be the final version, but I like it better.

**Comment by WorldLanguages** [comment]
In places where we aren't constrained to 132 chars, I assume we would continue to use the full sentence we already agreed upon before:

> Scratch Addons combines new and existing features and themes for the Scratch website and project editor into one easy-to-access and configurable browser extension.

And, depending on the context, we would add "the mission is..." after that, or not. For example, we previously decided here that the mission sentence wouldn't fit on the scratchaddons.com frontpage first portion (the one you don't need to scroll in order to read). It would probably fit on the README.md file on this repository, however.

**Comment by WorldLanguages** [comment]
Another alternative under 132 chars could be:

`Scratch Addons combines features and themes for the Scratch website and project editor into one configurable extension.` (119 chars)

("new and existing" removed from the proposal with 17 upvotes, "into one customizable extension" added to the end)

**Comment by DNin01** [comment]
> Another alternative under 132 chars could be:
> 
> `Scratch Addons combines features and themes for the Scratch website and project editor into one configurable extension.` (119 chars)
> 
> ("new and existing" removed from the proposal with 17 upvotes, "into one customizable extension" added to the end)

That could work, possibly modified slightly.

**Comment by WorldLanguages** [comment]
Alright, we can discuss whether to change the extension description in a future moment.

This new mission statement would then be used in places such as:
1. ScratchAddons.com homepage (only the first sentence of the two)
2. ScratchAddons.com FAQ
3. READMEs on GitHub, such as the org readme and this repo's README
4. Potentially YouTube channel description, Discord server information, etc.

And we should change them as soon as we can.

**Comment by penguinmoose** [comment]
> Alright, we can discuss whether to change the extension description in a future moment.
> 
> This new mission statement would then be used in places such as:
> 
> 1. ScratchAddons.com homepage (only the first sentence of the two)
> 2. ScratchAddons.com FAQ
> 3. READMEs on GitHub, such as the org readme and this repo's README
> 4. Potentially YouTube channel description, Discord server information, etc.
> 
> And we should change them as soon as we can.

We can put changing the extension description in a separate issue.

**Comment by Hans5958** [comment]
Can I have a recap for this? https://github.com/ScratchAddons/.github/blob/master/profile/README.md already have the new statement on it, but I want to see the final result of it.

**Comment by WorldLanguages** [comment]
@Hans5958 Not sure what you meant on March. What's the "final result" you're looking for?

**Comment by Hans5958** [comment]
The agreed mission statement. There have been alternatives as I saw but I just want to make sure which one is the final, agreed one. Is it the one on https://github.com/ScratchAddons/ScratchAddons/issues/5211#issuecomment-1341494126?

**Comment by WorldLanguages** [comment]
@Hans5958 Yes.

**Comment by Hans5958** [comment]
I see. I just read that for the landing intro, it would use the first sentence.

> We can put changing the extension description in a separate issue.

Has an issue been made? If it is then we can close this issue because we already reached consensus about the mission statement.

**Comment by WorldLanguages** [comment]
> > We can put changing the extension description in a separate issue.
> 
> Has an issue been made?

Not yet.

**Comment by WorldLanguages** [comment]
> I see. I just read that for the landing intro, it would use the first sentence.

Possibly both sentences also fit. I wouldn't say the decision to only use the first sentence for the landing intro is final.
